frontend_tests/casper_tests/00-realm-creation.js

```diff
@@ -0,0 +1,91 @@
+var common = require('../casper_lib/common.js').common;
+
+var email = 'alice@example.com';
+var domain = 'example.com';
+var organization_name = 'Awesome Organization';
+
+casper.start('http://localhost:9981/create_realm/');
+
+casper.then(function () {
+    // Submit the email for realm creation
+    this.waitForSelector('form[action^="/create_realm/"]', function () {
+        this.fill('form[action^="/create_realm/"]', {
+            email: email
+        }, true);
+    });
+    // Make sure confirmation email is send
+    this.waitWhileSelector('form[action^="/create_realm/"]', function () {
+         var regex = new RegExp('^http:\/\/[^\/]+\/accounts\/send_confirm\/' + email);
+         this.test.assertUrlMatch(regex, 'Confirmation mail send');
+    });
+});
+
+// Special endpoint enabled only during tests for extracting confirmation key
+casper.thenOpen('http://localhost:9981/confirmation_key/');
```

Comment:
Is there a not good way to have Casper do this as an AJAX request rather than having this navigate to that URL?  Not a big deal for this thing, but it seems like it could be good to have this sort of thing be in our toolkit.